lib/Graph/Graph.cpp

```diff
@@ -338,6 +348,23 @@ Variable *Module::createVariable(ElemKind T, llvm::ArrayRef<size_t> dims,
   return createVariable(FT, name, visibility, train, val);
 }
 
+Variable *
+Module::createSingletonVariable(ElemKind T, llvm::ArrayRef<size_t> dims,
+                                float scale, int32_t offset,
+                                llvm::StringRef name, VisibilityKind visibility,
+                                Variable::TrainKind train, float val) {
+  for (auto *V : getVars()) {
+    if (V->getName() == name) {
+      assert(false && "Variable with this name has already been created.");
+    }
+  }
+
+  auto FT = uniqueType(T, dims, scale, offset);
```

Comment:
Once we checked that a variable with a given name does not exist, can't we delegate to the usual `createVariable` instead of duplicating the logic here?

Reply:
`createVariable()` calls `addVar()` which uniquifies the name of the Variable. Here, we avoid the uniquify. The name has already been added in the Module constructor to `uniqueVariableNames_ ` to ensure it's not taken by another variable. So if we go through `createVariable()` the singleton variable will be given a different (unique) name, and future calls to `getVariableByName()` would not find the singleton.

Reply:
Oh, I see.